hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/RecordFinalizerBase.java

```diff
@@ -185,7 +185,11 @@ protected Map<TokenID, List<NftTransfer>> nftChangesFrom(
             // will be explicitly set as 0.0.0
             final var builder = NftTransfer.newBuilder();
             if (modifiedNft != null) {
-                builder.receiverAccountID(modifiedNft.ownerId());
+                if (modifiedNft.hasOwnerId()) {
+                    builder.receiverAccountID(modifiedNft.ownerId());
+                } else {
+                    builder.receiverAccountID(token.treasuryAccountId());
```

Comment:
I saw this change in one other PR also from Limechain 🤔  I believe we need to set sender to 0.0.0 if persistedNft is null on line 182

Reply:
Done